experimental/harfbuzz/src/lib.rs

```diff
@@ -147,6 +148,27 @@ unsafe extern "C" fn icu4x_hb_unicode_mirroring_destroy(user_data: *mut c_void)
     let _ = Box::from_raw(user_data as *mut BidiAuxiliaryProperties);
 }
 
+struct ScriptDataForHarfBuzz {
+    script_map: CodePointMapData<Script>,
+    enum_to_name_mapper: PropertyEnumToValueNameMapper<Script>,
+}
+
+unsafe extern "C" fn icu4x_hb_unicode_script(
+    _ufuncs: *mut hb_unicode_funcs_t,
+    unicode: hb_codepoint_t,
+    user_data: *mut c_void,
+) -> hb_script_t {
+    let script_data: &ScriptDataForHarfBuzz = &*(user_data as *mut ScriptDataForHarfBuzz);
+    let script: Script = script_data.script_map.as_borrowed().get32(unicode);
+    let enum_to_name_mapper = script_data.enum_to_name_mapper.as_borrowed();
+    let name: &str = enum_to_name_mapper.get(script).unwrap_or("Zzzz");
+    hb_script_from_string(name.as_ptr() as *const i8, name.len().try_into().unwrap_or(0))
```

Comment:
Good point about i8

But since HB gives us this nice function to call, shouldn't we call it? It basically does the cast but it does it more safely. If HB is statically linked I think/hope it is inlined.

Reply:
> It's a bit unfortunate that these ended up not being TinyAsciiStr<4>.

Yeah, well it got added as a generic API and data structure to support all properties, whose short names are not always 4 letters.

But I'm not opposed to making a special structure for this case since it would be more efficient in storage.

Reply:
> But since HB gives us this nice function to call, shouldn't we call it?

Because it does unnecessarily much. The HB call is meant for the case where you have an arbitrary ISO tag: perhaps not in canonical case (upper case and three lower-case letters) and perhaps not the canonical tag for each case. When we know that we're mapping our enum to the most canonical 4-byte string, we know that the 4 bytes can be stuffed in a 32-bit integer without branches without having to account for case variation or for not having the most canonical tag.

AFAICT, calling the HB function results in an extra function call and a bunch of branches that buy us nothing.

Reply:
Let's merge what we have here and then fix the rest in https://github.com/unicode-org/icu4x/issues/3256